api/internal/git/repospec_test.go

```diff
@@ -105,6 +112,26 @@ func TestNewRepoSpecFromUrlErrors(t *testing.T) {
 			"https://github.com/org/repo.git//path/../../exits/repo",
 			"url path exits repo",
 		},
+		"bad github separator": {
+			"github.com!org/repo.git//path",
+			"url lacks host",
+		},
+		"mysterious gh: prefix previously supported is no longer handled": {
+			"gh:org/repo",
+			"url lacks orgRepo",
+		},
+		"username unsupported with http": {
```

Comment:
Note that all three of these would be accepted on master, so the errors here are a behaviour change. I'm not 100% sure about this yet; will ponder more before we merge.

Reply:
Good point. I updated the code and tests to allow it. As commented elsewhere, I also decided against this validation more generally.